### DISCUSSION TITLE
Initializing Supabase For Unit Tests (Local Development)

### DISCUSSION THREAD

**Discussion by angus-clark** [supabase/supabase#11309] [Questions]
I am trying to write some unit tests for my Flutter Supabase queries but I am running into an error with path_provider whilst initializing the client. Here is what I am trying to run for my flutter tests:
```
void main() {
  TestWidgetsFlutterBinding.ensureInitialized();

  setUp(() async {
    await dotenv.load(fileName: Environment.fileName);
    await Supabase.initialize(
      url: Environment.supabaseURL,
      anonKey: Environment.supabaseAnonKey,
      debug: true,
    );
  });

  test('First test', () async {
    
  });
}
```

And here is the error I am getting:
```
***** Supabase init completed Instance of 'Supabase'
package:flutter/src/services/platform_channel.dart 294:7  MethodChannel._invokeMethod
===== asynchronous gap ===========================
dart:async                                                _Completer.completeError
package:supabase_flutter/src/supabase_auth.dart 116:44    SupabaseAuth.initialize
===== asynchronous gap ===========================
dart:async                                                _CustomZone.registerBinaryCallback
package:supabase_flutter/src/supabase_auth.dart 80:7      SupabaseAuth.initialize
package:supabase_flutter/src/supabase.dart 86:24          Supabase.initialize
test/account/account_repository_tests.dart 18:20          main.<fn>
===== asynchronous gap ===========================
dart:async                                                _CustomZone.registerUnaryCallback
test/account/account_repository_tests.dart 17:5           main.<fn>
MissingPluginException(No implementation found for method getApplicationDocumentsDirectory on channel plugins.flutter.io/path_provider_macos)
```

I've tried doing `flutter clean` with no result. Any help would be appreciated. Or perhaps there is a better way to do unit testing in supabase? I don't want to mock the supabase client because I want to test my queries e.g. `supabase.from('my_table').select('...')` are correct.

**Comment by yurifrl** [comment]
Reviving this thread, so apologies in advance. Here's the lowdown:

I tried various ways, but it seems the best shot to run Supabase.initialize in a test is by using a mock, like the one [here](https://github.com/supabase/supabase-flutter/blob/main/packages/supabase_flutter/test/widget_test_stubs.dart).

Started off like the original poster, trying a direct call, but ran into the same issues. Tried to tackle those errors, but it got pretty convoluted. Felt like maybe that wasn't the right approach.

I then gave wrapping the initializer in a widget a go, but ended up with the same errors.

The mock method worked out, setting some things up and passing a couple of arguments did the trick. However, I'm still curious about running it without the mock. that's arguably a naive approach to testing, I can't justify it but, I will probably keep looking for a workaround.
